Client/NoOpRunner.Client.Logic/Commands/ChangeVolumeCommand.cs

```diff
@@ -0,0 +1,40 @@
+﻿using NoOpRunner.Client.Logic.ViewModels;
+
+namespace NoOpRunner.Client.Logic.Commands
+{
+    public class ChangeVolumeCommand : BaseCommand<int>
+    {
+        private readonly SettingsViewModel settingsViewModel;
+
+        private int previousVolume = 0;
+
+        public ChangeVolumeCommand(SettingsViewModel settingsViewModel)
+        {
+            this.settingsViewModel = settingsViewModel;
+        }
+
+        protected override bool ExecuteInternal(int newVolume)
+        {
+            previousVolume = settingsViewModel.VolumeLevel;
+            settingsViewModel.VolumeLevel = newVolume;
+
+            return true;
+        }
+
+        protected override bool PreExecute(int newVolume)
+        {
+            if (settingsViewModel == null)
+                return false;
+
+            if (newVolume < 0)
+                return settingsViewModel?.VolumeLevel - newVolume >= 0;
+
+            return settingsViewModel?.VolumeLevel + newVolume <= 100;
```

Comment:
Here you increase by `newVolume`, but int execute you just set the volume to `newVolume`? Shouldn't these checks just be 0 <= newVolume <= 100?

Reply:
Fixed